python/ray/scripts/scripts.py

```diff
@@ -409,22 +409,34 @@ def stop():
     # Note that raylet needs to exit before object store, otherwise
     # it cannot exit gracefully.
     processes_to_kill = [
-        "raylet",
-        "plasma_store_server",
-        "raylet_monitor",
-        "monitor.py",
-        "redis-server",
-        "default_worker.py",  # Python worker.
-        " ray_",  # Python worker.
-        "org.ray.runtime.runner.worker.DefaultWorker",  # Java worker.
-        "log_monitor.py",
-        "reporter.py",
-        "dashboard.py",
+        # The first element is to filter ps results by command name (only the
+        # executable name). The second element is to filter ps results by
+        # command with all its arguments. See STANDARD FORMAT SPECIFIERS
+        # section of http://man7.org/linux/man-pages/man1/ps.1.html about comm
+        # and args. The two levels of filters can help avoid killing non-ray
+        # processes.
+        ["raylet", None],
+        ["plasma_store_server", None],
+        ["raylet_monitor", None],
+        [None, "monitor.py"],
+        ["redis-server", None],
+        [None, "default_worker.py"],  # Python worker.
+        [" ray_", None],  # Python worker.
```

Comment:
I only found that `ray_worker` matches this rule. Anyone can tell me the other possible processes? And why there's a whitespace at the beginning? In what case the whitespace is needed?

Reply:
Actor processes also match `ray_`. But I don't know why the whitespace is needed. 
@ericl do you know why? this is added by #3219.